### DISCUSSION TITLE
Autocomplete moves cursor to the next line.

### DISCUSSION THREAD

**Discussion by nd7141** [microsoft/WSL#6164] [General] [answered]
# Environment

```none
Windows build number: Version 10.0.18362.1139
Your Distribution version: 20.04
Whether the issue is on WSL 2 and/or WSL 1: Linux version 4.4.0-18362-Microsoft (Microsoft@Microsoft.com) (gcc version 5.4.0 (GCC) ) #1049-Microsoft Thu Aug 14 12:01:00 PST 2020
```

# Steps to reproduce

When I connect to the remote environment via WSL and try to use autocomplete, my cursor moves to the next line and there is a visible break in the path, which hinders readability. See below. 

![image](https://user-images.githubusercontent.com/4607709/97460322-effd0200-193c-11eb-85de-89444e1a978f.png)

Is it a known issue? How can I fix it? What it can be related to?

**Comment by Biswa96** [comment]
Is that in Command Prompt? Try to use a terminal emulator like mintty, putty, ConEmu etc.

**Comment by nd7141** [comment]
Can you please elaborate? I connect via WSL to a remote machine.

**Comment by therealkenc** [comment]
> Can you please elaborate?

He means try an alternate terminal emulator for WSL (the client side) to eliminate the Windows Console or Windows Terminal as the variable. That's probably not it in this instance tho.

That newline (line break) is expected behavior. You've got two files that start with `ex`, so the first thing bash completion is going to do is insert a newline. What is unexplained is why you aren't getting _both_ `example.txt` and `example2.txt` back from the remote, just `ample`. Like:

![image](https://user-images.githubusercontent.com/7049700/97502991-74e21e80-1931-11eb-9e06-0b14189e7170.png)

As a sanity check, connect to your remote with the [win32 OpenSSH](https://docs.microsoft.com/en-us/windows-server/administration/openssh/openssh_overview) client and see if autocomplete works as expected.

**Comment by nd7141** [comment]
Yes, it works for other terminals. For example, here for Cygwin. 

![image](https://user-images.githubusercontent.com/4607709/97718227-fd91c380-1ac5-11eb-8022-470df542305e.png)

**Comment by nd7141** [comment]
What' annoying about this issue is that when I tab, it moves the cursor to the end of the line, so that if I want to erase the last letters of the path, I will need to do delete a lot space before. 

![image](https://user-images.githubusercontent.com/4607709/97718581-62e5b480-1ac6-11eb-97d0-bfcf89750ff3.png)

**Comment by therealkenc** [comment]
As a sanity check, connect to your remote with the [win32 OpenSSH](https://docs.microsoft.com/en-us/windows-server/administration/openssh/openssh_overview) client from a `cmd.exe` prompt and see if autocomplete works as expected.

WAG might be termcap related. Once you are in the remote, try:
```
$ export TERM=vt100
```

That probably won't work, but is cheap to try. The other thing you can try is installing [18.04](https://www.microsoft.com/en-us/p/ubuntu-1804-lts/9n9tngvndl3q) from the store, and do the `ssh` before any local customizations. Start it with `wsl.exe -d Ubuntu-18.04`. There is nothing special about 18.04 here; its just a way to get a clean install with another name.

**Comment by nd7141** [comment]
So, logging to remote from wins32 ssh and doing autocomplete does not work. 
Cygwin works. 

However, I indicated that the problem appears only when I use WSL (or win32) in screen mode. Cygwin in screen works just fine. 

```
export TERM=vt100
```
also didn't help.

**Comment by nd7141** [comment, marked as answer]
Okay, finally fixed this by adding `term xterm-256color` to .screenrc file. From [this solution](https://stackoverflow.com/questions/52259357/linux-screen-tab-misbehaves-on-wsl-environment#).
